motr/client.h

```diff
@@ -896,6 +904,75 @@ struct m0_config {
 	m0_bcount_t mc_addb_size;
 };
 
+/* Constants for protection info type, max types supported is 8 */
+enum
+{
+	M0_PI_TYPE_MD5,
+	M0_PI_TYPE_MD5_INC_DIGEST,
+	M0_PI_TYPE_CRC,
+	M0_PI_TYPE_MAX
+};
+
+enum m0_pi_calc_flag {
+
+	/* No special handling */
+	M0_PI_CALC_NONE = 0,
+	/* PI calculation for data unit 0 */
+	M0_PI_CALC_UNIT_ZERO = 1 << 0
+};
+
+M0_BASSERT(M0_PI_TYPE_MAX <= 8);
+
+struct m0_pi_hdr {
+	/* type of protection algorithm being used */
+	uint8_t pi_type : 3;
+        /*size of PI Structure in multiple of  32 bytes*/
+        uint8_t pi_size : 5;
+};
+
+struct m0_md5_pi {
+
+#define PI_VALUE_LENGTH 16
+
+	/* header for protection info */
+	struct m0_pi_hdr hdr;
+	/* protection value computed for the current data*/
+	unsigned char pi_value[PI_VALUE_LENGTH];
+	/* structure should be 32 byte aligned */
+	char pad[M0_CALC_PAD(sizeof(struct m0_pi_hdr)+ PI_VALUE_LENGTH, 32)];
+};
+
+struct m0_md5_inc_digest_pi {
+
+#define PI_DIGEST_LENGTH 92
+#define PI_VALUE_LENGTH 16
+
+	/* header for protection info */
+	struct m0_pi_hdr hdr;
+	/*digest of previous data unit, required for checksum computation */
+	unsigned char prev_digest[PI_DIGEST_LENGTH];
+	/* protection value computed for the current data unit.
+	 * This checksum is calculated without seed
```

Comment:
If seed is not provided then this checksum is calculated without seed

Reply:
Done.